cli/cmd/plugin/standalone-cluster/cluster/cluster.go

```diff
@@ -33,6 +33,9 @@ type ClusterManager interface {
 	Get(clusterName string) (*KubernetesCluster, error)
 	// Delete will destroy a cluster or return an error indicating a problem.
 	Delete(c *config.LocalClusterConfig) error
+	// PullImage will fetch an image. This can be used to prefetch before calling
+	// create in order to get better feedback on the current activity.
+	PullImage(imageName string) error
```

Comment:
Should we consider a more generic API here? Perhaps:

```go
Prepare(c *config.LocalClusterConfig) error
```

My fear with specific APIs like `PullImage` is that as more specific edge cases grow up, so does our interface. With will then mean that work will be required for any future provider?

cc @jpmcb

Reply:
I agree - a `Prepare` method in the `ClusterManager` interface would fit nicely, that way, other programmer "users" implementing this would be able to define how the management of their k8s clusters can be prepared. 

If we pass the `LocalClusterConfig`, then we can introspect the node image and download it early for the kind provider

Reply:
That makes sense to me.

The only challenge then is the console output. Since the CLI code is just asking the `ClusterManager` to prepare for cluster creation, we can't really be specific about "Pulling image" since that may not be relevant for the given manager (like the noop one). Any suggestions on what we should show as output for this step?

Reply:
This is a really solid and valid concern. We need to talk today about better ways to propagate implementations of our Logger interface into lower-level packages such that they can propagate details around progress, which can vary per-provider.

Another option is to provide channels where we pass messages around, while neat architecturally, I really don't dig when projects do this, it invokes too much complexity IMO when a simple, shared logger is wildly adequate.